CRM/Core/Report/Excel.php

```diff
@@ -101,6 +91,28 @@ public static function makeCSVTable($header, $rows, $titleHeader = NULL, $output
     }
   }
 
+  /**
+   * Output the header row for a csv file.
+   *
+   * @param array $header
+   *   Array of field names.
+   *
+   * @return array
+   */
+  public static function outputHeaderRow($header): array {
+    $schema_insert = '';
+    $separator = Civi::settings()->get('fieldSeparator');
+    foreach ($header as $field) {
+      $schema_insert .= '"' . str_replace('"', '""', stripslashes($field)) . '"';
+      $schema_insert .= $separator;
+    }
+    // end while
+    // need to add PMA_exportOutputHandler functionality out here, rather than
+    // doing it the moronic way of assembling a buffer
+    // We append a hex newline at the end.
+    echo trim(substr($schema_insert, 0, -1)) . "\015\012";
```

Comment:
addCharacter is just https://github.com/civicrm/civicrm-core/pull/16364/files#diff-aecd4a7f55b19f44ba714f8ac41b1bbcL43 - we can just put in the string